.github/workflows/pull-request-checks.yaml

```
@@ -60,6 +60,11 @@ jobs:
           cmake .. -G Ninja -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_COMPILER=/usr/bin/gcc -DCMAKE_CXX_COMPILER=/usr/bin/g++
       - name: Build with Ninja
         run: cd build; ninja
+      - name: Check if package building works
+        run: |
+          cd build
+          ninja package
+          ls *.deb
```

What does the `ls *.deb` achieve, other than a nice output? Will it fail the build in someway? Or will it mask any failures from the `ninja package` because the exit code of the `ls` will be success?

---

My experience of debugging the run sections of github actions suggests that it will stop on the first line where an error code is returned. Similar to writing a bash script where `set -e` has been specified.

---

@chrisr-diffblue `ls` returns a non-zero exit code if called on non-existent files... so this is just checking whether or not there is a file ending with `.deb` in the directory after `ninja package`.  See [here](https://github.com/hannes-steffenhagen-diffblue/github-actions-test/pull/4/checks?check_run_id=1011772782) for a demonstration of how this works.

Failures propagate in github action scripts because I believe `set -e` is set by default.